### PR TITLE
Show generic signature in error message

### LINKED ISSUE #480: Method arg mismatch error is not actionable
```
Error: chat_request() lacks `...` so method formals must match generic formals exactly
```

Would be helpful to include what it has and what it's supposed to have.

### PR DESCRIPTION
**Pull request by t-kalinowski**
closes #480

### CODE REVIEW

**Comment by hadley** [comment]
I think it also needs to show what you need, i.e. the args to the generic.

**Comment by t-kalinowski** [comment]
I might be misunderstanding, but it does show the generic args. Do you mean we should also show the "wrong" method args in the error message?

**Comment by hadley** [comment]
Oh sorry yeah, it should show both the generic and the method args.

File: R/method-register.R [hunk 1]
```diff
@@ -181,9 +181,14 @@ check_method <- function(method, generic, name = paste0(generic@name, "(???)"))
...
   if (!"..." %in% generic_args && !identical(generic_formals, method_formals)) {
     msg <- sprintf(
-      "%s lacks `...` so method formals must match generic formals exactly",
-      show_args(formals(generic), name = generic@name)
+      "%s() generic lacks `...` so method formals must match generic formals exactly.",
+      generic@name
     )
+    bullets <- c(
+      sprintf("- generic formals: %s", show_args(generic_formals, name = generic@name)),
+      sprintf("-  method formals: %s", show_args(method_formals, name = generic@name))
```
Comment (hadley) [inline R/method-register.R:189]:
```suggestion
      sprintf("- method formals: %s", show_args(method_formals, name = generic@name))
```

Comment (hadley) [reply R/method-register.R:189]: That makes the alignment look better to me.

**Review by hadley — APPROVED** [review]
Perfect, thanks!

**Pull request merged**